src/sicmutils/abstract/function.cljc

```diff
@@ -166,22 +166,26 @@
          :else
          (u/illegal (str "WTF range" range)))))
 
+(defn binding-pairs [litfns]
```

Comment:
I pulled the logic that lived outside of quasiquote land so that we had a bit less to duplicate in the sci macros namespace. I tested that this works, and the existing tests will exercise it too.